src/containers/study/StudyDetails.js

```diff
@@ -145,19 +151,37 @@ type Props = {
 const StudyDetails = ({ study, notificationsEnabled } :Props) => {
   const dispatch = useDispatch();
   const [editModalVisible, setEditModalVisible] = useState(false);
+  const [isDeleteModalVisible, showDeleteModal, hideDeleteModal] = useBoolean(false);
+
+  const deleteStudyRS = useRequestState(['studies', DELETE_STUDY]);
 
   const studyDescription = study.getIn([STUDY_DESCRIPTION, 0]);
   const studyUUID = study.getIn([STUDY_ID, 0]);
   const studyVersion = study.getIn([STUDY_VERSION, 0]);
   const studyEmail = study.getIn([STUDY_EMAIL, 0]);
   const studyGroup = study.getIn([STUDY_GROUP, 0]);
 
+  useEffect(() => {
+    if (deleteStudyRS === RequestStates.SUCCESS) {
+      setTimeout(() => {
+        dispatch(removeStudyOnDelete(studyUUID));
+        dispatch(goToRoute(Routes.ROOT));
+        dispatch(resetRequestState(DELETE_STUDY));
+      }, 2000);
```

Comment:
Why did you decide to go with `setTimeout` instead of handling this logic in the reducer?

Reply:
`setTimeout` gives the user an opportunity to see a successful deletion status message.